src/System.Reflection.Metadata.ILDasm/src/Decoder/ILDecoder.cs

```diff
@@ -0,0 +1,582 @@
+﻿using ILDasmLibrary.Instructions;
+using Microsoft.CSharp;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Decoding;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+
+namespace ILDasmLibrary.Decoder
+{
+    public struct ILDecoder
+    {
+        #region Public APIs
+
+        /// <summary>
+        /// Method that given a token defines if it is a type reference token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a type reference false if not</returns>
+        public static bool IsTypeReference(int token)
+        {
+            return (token >> 24) == 0x01;
```

Comment:
The ```token >> 24``` shifting occurs a fair amount.  Should that be factored out into some private method that gives this an appropriate name?